sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ConnectionPolicy.java

```diff
@@ -96,11 +96,14 @@ private ConnectionPolicy(
                 .DirectConnectionConfigHelper
                 .getDirectConnectionConfigAccessor()
                 .isHealthCheckTimeoutDetectionEnabled(directConnectionConfig);
+
+        // NOTE: should be compared with COSMOS.MIN_CONNECTION_POOL_SIZE_PER_ENDPOINT
+        // read during client initialization before connections are created for the container
         this.minConnectionPoolSizePerEndpoint =
-                ImplementationBridgeHelpers
-                        .DirectConnectionConfigHelper
-                        .getDirectConnectionConfigAccessor()
-                        .getMinConnectionPoolSizePerEndpoint(directConnectionConfig);
+                Math.max(ImplementationBridgeHelpers
```

Comment:
Why are we taking Math.max() everywhere?

Reply:
The idea is to fallback to the system property if it is set to a higher value.